### ISSUE TITLE
Rendering is not fast and smooth (OsmAnd Android)

### ISSUE THREAD

**Issue by xfrisakbaeun** [osmandapp/OsmAnd#14367]
### Description
The OsmAnd Android version is much slower and not smooth compared to the OsmAnd iOS version. The rendering in the iOS version is much faster and smoother.

Note:
On iOS I also tested Google Maps and Apple Maps and I can say that OsmAnd iOS is close to speed and smoothness in most situations.

### How to reproduce?
1. Use only current high-end smartphones.
2. Open OsmAnd Android and iOS for comparison.
3. Download the maps offline.
4. Zoom to the world map and zoom to any country from USA for example.
5. Try to zoom in and out and move the map.
6. You can clearly see that rendering on iOS is much faster and smoother than on Android.

### Your Environment
OsmAnd Version: 4.2.0
Android/iOS version: Android 12 and iOS 15
Device model: Xiaomi 12 Pro, Samsung Galaxy S22 and iPhone 13 Pro Max

**Maps used (online or offline):**   
offline

**Comment by julianfairfax** [comment]
@vshcherb I saw your comment on https://github.com/osmandapp/OsmAnd/issues/13448#issuecomment-1002830257 about this being a work in progress. Has any progress been made since then?

**Comment by xfrisakbaeun** [comment]
> New 4.2.0 build has OpenGL rendering available in development settings. Work in progress.

I just tested this in new 4.2.0 build (OpenGL) and there are no improvements, the iOS version is much smoother.

In addition, I also noticed that the general operation, whether opening settings, activating the search and other options that the app gives, the animations are not smooth on Android version. It often happens that there are small lags and does not feel like a native Android app. Can you improve this as well?

I highly recommend comparing an example to Google Maps, Google Maps on iOS and Android are smoother than OsmAnd. Would like to see this quality for OsmAnd app as well.

**Comment by julianfairfax** [comment]
> 

Yeah I didn't see much of an improvement when switching on the OpenGL option. Maybe that's just me? Or maybe it's the fact that I have a Pixel 3a? It felt the same, at least for me, and at least on my device.

**Comment by xfrisakbaeun** [comment]
> > 
> 
> Yeah I didn't see much of an improvement when switching on the OpenGL option. Maybe that's just me? Or maybe it's the fact that I have a Pixel 3a? It felt the same, at least for me, and at least on my device.

It's not a bug with your devices, I tested a Xiaomi 12 Pro and Samsung Galaxy S22 Ultra in 120 fps and 60 fps mode, same problem. The hardware here is up to date and one of the fastest ARM smartphones.

**Comment by julianfairfax** [comment]
> Switch requires restart of the application. OpenGL rendering is visually different than old one: opengl renders map with tiles (squares) if you don't see squares than it's not OpenGL version.
> 
> With S21 Ultra rendering has no visual artefacts while sliding.

I did that. Here are two screen recordings:
- https://send.vis.ee/download/fca231de90dd78e8/#NvlvoQUnbAXR0ppP0FWDkA
- https://send.vis.ee/download/6d88cc9e8835ddfb/#yMGvZ5-Ec4tbsKweLoXcyA

Both are from a Pixel 3a running Android 12 and using the latest version of the nightly build.

The first one is without OpenGL rendering enabled, and the second one is with it enabled.

The only map downloaded is of Berlin, which is not where I am and the location permission wasn't granted to OsmAnd.

I can certainly telling there is a difference in how the map is rendered, but I can't say that one is better than the other.

Obviously I have an old device, and I'm also running other apps in the background. But none were actually open at the same time, and this has been tested on newer devices as well.

**Comment by vshcherb** [comment]
As I see neither of these videos is OpenGL rendering. I can't see grid and tiles. Here is slow example of new rendering engine where it's noticeable grid & tiles.

In case you are using nightly build it's not included, only in beta.

https://user-images.githubusercontent.com/1042025/170741326-53d652e1-2ef2-4927-84ec-18641468928f.mp4

**Comment by julianfairfax** [comment]
> As I see neither of these videos is OpenGL rendering. I can't see grid and tiles. Here is slow example of new rendering engine where it's noticeable grid & tiles.
> 
> In case you are using nightly build it's not included, only in beta.
> 2022-05-27.18.33.44.mp4

Well there's an option for it in the development settings nonetheless, but yes, I was using the nightly build. Also, that is still not the best performance in my opinion.

**Comment by vshcherb** [comment]
You can post a video of your OpenGL rendering and I could say if it's our expected level or not. The video above is twice slower than expected. 
This is nightly build with opengl - https://download.osmand.net/latest-night-build/OsmAnd-opengl-arm64-nightly.apk

**Comment by julianfairfax** [comment]
> You can post a video of your OpenGL rendering and I could say if it's our expected level or not. The video above is twice slower than expected. 
> This is nightly build with opengl - https://download.osmand.net/latest-night-build/OsmAnd-opengl-arm64-nightly.apk

Oh, I didn't see that build on the website. Anyway, here's a video from that build without OpenGL rendering enabled: https://send.vis.ee/download/51480e5a1047e95e/#u7qQ8cvudXrRBozBDOHkRA, with OpenGL rendering enabled: https://send.vis.ee/download/6c02f16500041a44/#dqeYYg3acae49h1eXQQ6xA, and one from Organic Maps for comparison: https://send.vis.ee/download/f3f8c7428723a49e/#RUkkDOAZ2FDPdlSUGPGuxQ

There's definitely a difference with OpenGL rendering enabled, and it might even be better, but regardless, the experience still doesn't qualify as being smooth.

**Comment by vshcherb** [comment]
@julianfairfax it's definitely as smooth as it supposed to be. I can't believe it's S22 Ultra, weird.. . Probably it has more details than it should be. We will investigate further this version after release.

**Comment by julianfairfax** [comment]
> @julianfairfax it's definitely as smooth as it supposed to be. I can't believe it's S22 Ultra, weird.. . Probably it has more details than it should be. We will investigate further this version after release.

This device is a Pixel 3a

**Comment by julianfairfax** [comment]
It's starting to appear to me that, at least on my device, OsmAnd's OpenGL rendering is actually a lot better, but the nightly app itself is just a lot more unstable
